openwisp_controller/config/utils.py

```diff
@@ -78,6 +80,15 @@ def update_last_ip(device, request):
         update_fields.append("management_ip")
     if update_fields:
         device.save(update_fields=update_fields)
+    # When update fields are present then save() will run the WHOIS
+    # lookup. But if there are no update fields, we still want to
+    # trigger the WHOIS lookup if there is no record for the device's
+    # last_ip.
+    elif (
+        app_settings.WHOIS_CONFIGURED
+        and not device.whois_service.get_device_whois_info()
+    ):
+        device.whois_service.trigger_whois_lookup()
```

Comment:
I am not sure I like putting this code here. This function is only concerned with updating the IP fields.

Why don't we change the code which calls this function instead? If it returns false we can trigger the whois lookup.